### PR TITLE
[extension_google_sign_in_as_googleapis_auth] Move from flutter/plugins.

### PR DESCRIPTION
**Pull request by ditman**
This package doesn't make much sense in flutter/plugins, since it's not a true plugin, only an optional extension on google_sign_in.

Testing infrastructure has a hard time in flutter/plugins with this package.

It seems that flutter/packages might be a better home for the extension.

This PR moves the package to flutter/packages, and only minimally changes code to comply with the slightly different analysis rules of this repo.

## Pre-launch Checklist

- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated `CHANGELOG.md` to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the `#hackers-new` channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

**Comment by stuartmorgan** [comment]
> to comply with the slightly different analysis rules of this repo

Something I'm going to work on fixing :)

File: packages/extension_google_sign_in_as_googleapis_auth/pubspec.yaml [hunk 1]
```diff
@@ -6,22 +6,21 @@
...
+version: 3.0.0
+repository: https://github.com/flutter/packages/tree/master/packages/extension_google_sign_in_as_googleapis_auth
 
 dependencies:
   flutter:
     sdk: flutter
   google_sign_in: ^5.0.0
-  googleapis_auth: ^1.0.0
-  meta: ^1.3.0
+  googleapis_auth: ^1.1.0
```
Comment (stuartmorgan) [inline packages/extension_google_sign_in_as_googleapis_auth/pubspec.yaml:16]: Do we actually needed anything from the latest release, or can we leave 1.0.0 to give clients more flexibility?

File: packages/extension_google_sign_in_as_googleapis_auth/pubspec.yaml [hunk 2]
```diff
@@ -6,22 +6,21 @@
...
 
 dev_dependencies:
-  pedantic: ^1.10.0
-  test: ^1.16.3
   flutter_test:
     sdk: flutter
+  pedantic: ^1.10.0
 
 environment:
   sdk: ">=2.12.0-259.9.beta <3.0.0"
```
Comment (stuartmorgan) [inline packages/extension_google_sign_in_as_googleapis_auth/pubspec.yaml:26]: We could remove the prerelease bit now.

File: packages/extension_google_sign_in_as_googleapis_auth/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,8 @@
+## 3.0.0
```
Comment (stuartmorgan) [inline packages/extension_google_sign_in_as_googleapis_auth/CHANGELOG.md:1]: Why is this a breaking change? Nothing should affect clients in a breaking way.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

File: packages/extension_google_sign_in_as_googleapis_auth/pubspec.yaml (hunk 1 above)
Comment (ditman) [reply packages/extension_google_sign_in_as_googleapis_auth/pubspec.yaml:16]: This is only needed for the new import to be true.

I've also bumped the major version of this package to 3.0.0 so people have to opt-in to get the latest changes.

File: packages/extension_google_sign_in_as_googleapis_auth/pubspec.yaml (hunk 2 above)
Comment (ditman) [reply packages/extension_google_sign_in_as_googleapis_auth/pubspec.yaml:26]: Ah yes, good point!

File: packages/extension_google_sign_in_as_googleapis_auth/pubspec.yaml (hunk 1 above)
Comment (stuartmorgan) [reply packages/extension_google_sign_in_as_googleapis_auth/pubspec.yaml:16]: What changes are they opting into though? If it's just this, the resolver will handle cases where someone is pinning an old googleapis_auth for some reason, I just thought we didn't need to narrow the range (I missed that the new import was added in 1.1.0, vs it already having been there before the declaration was added).

File: packages/extension_google_sign_in_as_googleapis_auth/CHANGELOG.md (hunk 3 above)
Comment (ditman) [reply packages/extension_google_sign_in_as_googleapis_auth/CHANGELOG.md:1]: This used to be 2.0.1 until I changed the version of googleapis_auth to 1.1.0 for the new import.

File: packages/extension_google_sign_in_as_googleapis_auth/pubspec.yaml (hunk 1 above)
Comment (ditman) [reply packages/extension_google_sign_in_as_googleapis_auth/pubspec.yaml:16]: Nah, I sometimes use major version changes for people to check the changelog (so they see the new location of the package), but I guess this is 2.0.1. 2.0.0 tests still pass, so this change should be minor from our standpoint, I think.

Comment (stuartmorgan) [reply packages/extension_google_sign_in_as_googleapis_auth/pubspec.yaml:16]: > so they see the new location of the package

I would expect that ~0 people who aren't us will care where the source lives :)

Comment (ditman) [reply packages/extension_google_sign_in_as_googleapis_auth/pubspec.yaml:16]: Haha fair enough, let me get this back to 2.0.1 and remove the prerelease bit from the sdk version :P

**Comment by ditman** [comment]
`googleapis: ^2.0.0` with null safety has come out, so I've used this opportunity to update the missing parts of the example app, which is now **💪 Running with sound null safety 💪**

File: .metadata [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# This file tracks properties of this Flutter project.
+# Used by Flutter tool to assess capabilities and perform upgrades etc.
+#
+# This file should be version controlled and should not be manually edited.
```
Comment (ditman) [inline .metadata:4]: Should it? I don't remember consciously adding this file :/

**Comment by ditman** [comment]
**I'll merge this via the command-line, so the history of the package is preserved**, instead of squashing everything in a single commit.

**Review by stuartmorgan — APPROVED** [review]
LGTM

**Pull request merged**